torchgeo/trainers/cyclone.py

```diff
@@ -198,6 +199,20 @@ def setup(self, stage: Optional[str] = None) -> None:
         storms from the training set (specifically, the latter parts of the storms) as
         well as some novel storms.
         """
+        self.all_train_dataset = TropicalCycloneWindEstimation(
+            self.root_dir,
+            split="train",
+            transforms=self.custom_transform,
+            download=False,
+        )
+
+        self.all_test_dataset = TropicalCycloneWindEstimation(
+            self.root_dir,
+            split="test",
+            transforms=self.custom_transform,
+            download=False,
+        )
```

Comment:
I get that it's nice to download a dataset in one function and instantiate it elsewhere, but doesn't that require us to rebuild the entire dataset index twice? Seems quite a bit slower.

Reply:
According to pytorchlightning docs you shouldn't save state in `prepare_data`, but should download datasets there. You should setup the dataset objects in `setup`.